File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.js
const readline = require("readline");

(async () => {
  let rl = readline.createInterface({ input: process.stdin });
  const lines = [];
  let counter = 4;
  for await (const line of rl) {
    lines.push(line.split(" ").map(Number));
    counter--;
    if (counter == 0) rl.close();
  }
  let matrix = new Array(101).fill(0).map(() => new Array(101).fill(0));

  for (const [startX, startY, endX, endY] of lines) {
    for (let i = startX; i < endX; i++) {
      for (let j = startY; j < endY; j++) {
        matrix[j][i] = 1;
      }
    }
  }

  let sum = 0;
  matrix.forEach((row) => {
    const rowSum = row.reduce((acc, cur) => {
      return acc + cur;
    }, 0);
    sum += rowSum;
  });

  console.log(sum);
})();

File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.py
lines=[]
for _ in range(4):
    lines.append(list(map(int, input().split())))

matrix=[[0 for _ in range(0, 101)] for _ in range(0, 101)]

for startX, startY, endX, endY in lines:
    for i in range(startX, endX):
        for j in range(startY, endY):
            matrix[j][i] = 1

answer=0
for line in matrix:
    answer+=sum(line)

print(answer)